lib/iex/lib/iex/introspection.ex

```diff
@@ -115,13 +126,37 @@ defmodule IEx.Introspection do
         end
         :ok
       else
-        :not_found
+        if has_callback?(mod, fun, arity) do
+          :behaviour_found
+        else
+          :not_found
+        end
       end
     else
       :no_docs
     end
   end
 
+  defp has_callback?(mod, fun) do
+    mod
+    |> Code.get_docs(:callback_docs)
+    |> find_callback_doc(fun)
+  end
+
+  defp has_callback?(mod, fun, arity) do
+    mod
+    |> Code.get_docs(:callback_docs)
+    |> find_callback_doc(fun, arity)
+  end
+
+  defp find_callback_doc(docs, fun) do
+    Enum.any?(docs, &match?({{^fun, _}, _, _, _}, &1))
+  end
+
+  defp find_callback_doc(docs, fun, arity) do
+    Enum.any?(docs, &match?({{^fun, ^arity}, _, _, _}, &1))
```

Comment:
Actually, nevermind. The current way is fine because it is at least in sync with the line above.